plugins/gradle/gradle-plugin/src/main/java/org/wildfly/swarm/plugin/gradle/ThorntailArquillianPlugin.java

```diff
@@ -44,11 +58,68 @@ public ThorntailArquillianPlugin(ToolingModelBuilderRegistry registry) {
     @Override
     public void apply(Project project) {
         super.apply(project);
-        // Add the Gradle tooling dependency if it is missing.
-        Configuration testRuntimeConfig = project.getConfigurations().findByName("testRuntimeClasspath");
-        if (testRuntimeConfig != null) {
-            project.getDependencies().add("testRuntimeClasspath",
-                                          "org.gradle:gradle-tooling-api:" + project.getGradle().getGradleVersion());
+
+        //noinspection Convert2Lambda
+        project.afterEvaluate(new Action<Project>() {
+            @Override
+            public void execute(Project __) {
+                final String CONFIGURATION = "testRuntimeClasspath";
+                // Add the Gradle tooling dependency if it is missing.
+                String gav = "org.gradle:gradle-tooling-api:" + project.getGradle().getGradleVersion();
+                Configuration config = project.getConfigurations().findByName("testRuntimeClasspath");
+                if (config != null) {
+                    DependencyHandler handler = project.getDependencies();
+                    handler.add(config.getName(), handler.create(gav));
+                } else {
+                    System.err.println("Unable to add Gradle Tooling APIs to the " + CONFIGURATION +
+                            ". Thorntail Arquillian integration might not work.");
+                }
+
+                URL resource = ThorntailArquillianPlugin.class.getClassLoader().getResource(BUILD_SCRIPT_FRAGMENT);
+                if (resource != null) {
+                    try {
+                        if (!project.getBuildDir().exists()) {
```

Comment:
Resolved a small bug that would have become a show stopper :) .. All thanks to the Thorntail Examples. I should probably add a note to the README reminding myself and others that we should verify the plugin changes against the example projects before raising a PR.